src/iframe-helper.js

```diff
@@ -381,10 +382,10 @@ function getSentinel_(iframe, opt_is3P) {
  */
 function parseIfNeeded(data) {
   const shouldBeParsed = typeof data === 'string'
-      && data.charAt(0) === '{';
+      && data.indexOf('amp-') == 0;
```

Comment:
Is this changed needed right now? No code except IframeMessagingClient is sending message in this format yet. I would expect a deprecation of this whole method in your refactoring CL.